### ISSUE TITLE
VS2017 RC - The target “GetCopyToPublishDirectoryItems” does not exist in the project.

### ISSUE THREAD

**Issue by srivatsn** [dotnet/sdk#543]
_From @Delie on November 18, 2016 23:17_

After installing Visual Studio 2017 RC, and upgrading my ASP.NET Core Web Application (.NET Framework) project to CSPROJ, I am no longer able to publish my project to the file system. Building the solution works fine, but when I publish I get this error:

> The target "GetCopyToPublishDirectoryItems" does not exist in the project.

To replicate:

1. Create a new ASP.NET Core Project (.NET Framework), and solution.
2. Add a new Class Library project (.NET Framework) to the solution.
3. Reference the Class Library project in the .NET Core project.
4. Instantiate a class from the Class Library project in the .NET Core project
5. Rebuild solution.
6. Publish the .NET Core project and it throws the error.

_Copied from original issue: aspnet/Tooling#869_

**Comment by srivatsn** [comment]
_From @discostu105 on November 23, 2016 19:52_

I have exactly the same problem.

btw, "GetCopyToPublishDirectoryItems" is defined over there: https://github.com/dotnet/sdk/blob/master/src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Publish.targets

Maybe @eerhardt knows something about that? Is there a workaround? I'm trying to port my project to .csproj, but this blocks me.

**Comment by srivatsn** [comment]
_From @duracellko on December 2, 2016 10:21_

I tried workaround to add empty target to referenced class library project:
<Target Name="GetCopyToPublishDirectoryItems"/>
It helped a bit, but then I encountered another error. It is mentioned in known issues. [https://github.com/aspnet/Tooling/blob/master/known-issues-vs2017.md](https://github.com/aspnet/Tooling/blob/master/known-issues-vs2017.md)
"Unable to publish"

I tried to find a workaround by overriding some targets. After few hours I gave up and I converted project to standard .NET Framework Console Application.

1. Delete old project (make a backup of course).
2. Create new project using template Classic Windows - Console applications.
3. Add same NuGet packages as in old project.
4. Add all *.cs files from old project.
5. Add references to class library project.
6. Open _WebApp_.csproj in text editor.
7. Add following element at the end of file.

```
  <Target Name="AfterBuild">
    <Copy SourceFiles="$(SolutionDir)\packages\Libuv.1.9.1\runtimes\win7-x64\native\libuv.dll" DestinationFolder="$(TargetDir)" />
  </Target>
```

8. Add following element into App.config

```
<runtime>
 <gcServer enabled="true" />
</runtime>
```

9. Add Web.config from old project.
10. In Web.config change attribute aspNetCore/processPath to name of console application executable (WebApp.exe) and change arguments to empty string.

If you use different configuration in development environment, I added command line argument for debug project -AspNetCore:Environment=Development, then in Program.Main method I transformed this command line arguments to environment variable ASPNETCORE_ENVIRONMENT.

If you use User Secrets, then add into .csproj file element `<UserSecretsId>YourSecretId</UserSecretsId>`. Then you have to add secrets.json file to you AppData profile. No support from Visual Studio in this.

**Comment by srivatsn** [comment]
_From @duracellko on December 2, 2016 11:2_

And I forgot to set Copy to Output Directory for files web.config and appsettings.json.

My project is just WebAPI, so no .cshtml file. So it's easy.

**Comment by srivatsn** [comment]
_From @hybridview on December 20, 2016 16:33_

This issue doesn't appear to have gotten any attention yet. Does anymore know why it happens? I'm trying to find a workaround without having to hack up my project file.

**Comment by srivatsn** [comment]
_From @jsgoupil on December 20, 2016 16:55_

@hybridview Rolling back to WebAPI2 was the best solution I have done so far.

**Comment by srivatsn** [comment]
_From @eerhardt on December 20, 2016 21:39_

Sorry, I've been on vacation and haven't been checking email or GitHub.  I just saw this now.

The problem is that one project uses the new Microsoft.NET.Sdk project system and the referenced class library project doesn't use the same project system.

As a workaround, I can think of the following options:

1. Convert the class library project to use the new Microsoft.NET.Sdk project system.
2. Add a "fake/mock" empty target to your class library project named "GetCopyToPublishDirectoryItems".  This is mostly a hack, but it should get by the error mentioned above. You may run into other issues once you get by this error.

@srivatsn - Is it a supported scenario to mix and match these different project systems across P2P references?  

/cc @nguerrera @dsplaisted @davkean

**Comment by srivatsn** [comment]
_From @davkean on December 20, 2016 21:47_

Yes this is supported. Either we dummy out GetCopyToPublishDirectoryItems or sniff that the target doesn't exist (not sure this is even possible)?

**Comment by srivatsn** [comment]
Yes referencing an existing project from a new one is supported.

**Comment by srivatsn** [comment]
_From @coonmoo on December 20, 2016 22:32_

I have the same problem with Service Fabric, Asp.Net Core and VS 2017 RC2.

To replicate:

1. Create a new Service Fabric project
2. Select ASP.NET Core Web
3. Add a new Class Library project (.NET Framework) to the solution.
4. Reference the Class Library project in the .NET Core project.
5. Instantiate a class from the Class Library project in the .NET Core project
6. Rebuild solution.
7. Publish the Service Fabric project

I'd love to see a fix for this, because VS 2017 is awesome.


```
Register-ServiceFabricApplicationType : The BuildLayout of the application in 
4>C:\SfDevCluster\Data\ImageBuilderProxy\AppType\SfApplication1Type is invalid. Code is missing for service Web1Pkg.
4>At C:\Program Files\Microsoft SDKs\Service 
4>Fabric\Tools\PSModule\ServiceFabricSDK\Publish-NewServiceFabricApplication.ps1:244 char:9
4>+         Register-ServiceFabricApplicationType -ApplicationPathInImage ...
4>+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
4>    + CategoryInfo          : InvalidOperation: (Microsoft.Servi...usterConnection:ClusterConnection) [Register-Servic 
4>   eFabricApplicationType], FabricException
4>    + FullyQualifiedErrorId : RegisterApplicationTypeErrorId,Microsoft.ServiceFabric.Powershell.RegisterApplicationTyp 
4>   e
4> 
4>Finished executing script 'Deploy-FabricApplication.ps1'.
4>Time elapsed: 00:00:12.3127633
4>The PowerShell script failed to execute.
```

**Comment by srivatsn** [comment]
_From @dsplaisted on December 21, 2016 1:27_

@srivatsn We should move this issue to the dotnet/sdk repo.  I don't think we currently have tests that cover building an SDK project which references a non-SDK project, much less publishing in this scenario.

**Comment by srivatsn** [comment]
_From @EuroForm on December 21, 2016 10:43_

Having same problem. Would also love to see a fix to this 😅🙏🔥

**Comment by NickNiebling** [comment]
I've got this problem in a .NET Framework 4.6.2 MVC project using "older class libraries" when using VS2017 RC. The website builds and run locally without any problems, but I'm unable to publish the website.

It's difficult for me to know exactly what makes the "older class libraries" work or not, since all my internal project references are having this issue in the solution.

Thanks for accepting the issue and looking into it 👍  Looking very much forward to a solution asap 😊

**Comment by jxlarrea** [comment]
What's the status on this issue? 

I am unable to publish my ASP.NET Core MVC application due to this. The workaround posted by @srivatsn does eliminate the "GetCopyToPublishDirectoryItems" error message but the web application fails at runtime, complaining about missing references "Can not find reference assembly '.NETFramework/v4.6.2/Microsoft.CSharp.dll' file for package microsoft.csharp". I'm sure this is caused by the lack of "refs" folder.

**Comment by derf26** [comment]
I have this issue too - can't publish, and now that the Web API project has been converted to .csproj, VS2015 can't load the project, so I can't go back to an older version of VS to publish it either. I'd rather not rollback the code changes I made since the upgrade... any workarounds would be appreciated also.

**Comment by souchprod** [comment]
Really looking for a fix too, make VS2017 RC2 quite unusable for several kind of projects ! Not easy for .net core projects to get back to VS 2015 when the 2017 is so much better :)

**Comment by cceuroform** [comment]
I have this issue too.
I found a "partial" solution.
I added the following to every referenced assembly
**<Target Name="GetCopyToPublishDirectoryItems"></Target>**

This made the problem go away, and the publish process continues much longer.
BUT now i have a different error

Error MSB3094: "DestinationFiles" refers to 1 item(s), and "SourceFiles" refers to 2 item(s). They must have the same number of items.	
C:\Users\cc\.nuget\packages\microsoft.net.sdk\1.0.0-alpha-20161104-2\build\Microsoft.NET.Publish.targets	126

**Comment by cceuroform** [comment]
I installed Build 26020.00, but still same issue

**Comment by drewid** [comment]
This is a really critical issue and should be really basic.  Just have a dotnetcore web project that is including a class library.  As people have said, building and running locally works fine, but publishing up to Azure gives the failure messages already mentioned.  This "really" sucks as spent a lot of time moving to VS 2017 and things have been pretty decent up until this bug bit me in the ass - at "build" time of all things.  Really frustrating as got everything else to work and build.

**Comment by agarwal-peeush** [comment]
Really looking for a fix too. I experience the same issue and unable to publish after upgrade.

**Comment by nguerrera** [comment]
For starters, here is a more complete workaround using a default target that makes non Microsoft.NET.Sdk-based projects respect the default publish semantics.

Add the following to a new or existing Directory.Build.targets somewhere that is above all of the projects in the solution (e.g.. repository root directory). MSBuild will then pick up the targets for any project underneath that directory:

``` XML
<Project>
 <!--
  Work around https://github.com/dotnet/sdk/issues/543 with a default implementation of  
  GetCopyToPublishDirectoryItems that simply returns whatever GetCopyToOutputDirectoryItems 
  does with CopyToPublishDirectory implied by CopyToOutputDirectory, which is the same as
  Microsoft.NET.Sdk default when CopyToPublishDirectory is not used. Microsoft.NET.Sdk projects 
  will override this to allow the publish output to be customized independently from the build output.
  -->
  <Target
      Name="GetCopyToPublishDirectoryItems" 
      DependsOnTargets="GetCopyToOutputDirectoryItems"
      Returns="@(AllPublishItemsFullPathWithTargetPath)">
      
    <ItemGroup>
      <AllPublishItemsFullPathWithTargetPath Include="@(AllItemsFullPathWithTargetPath)">
        <CopyToPublishDirectory>%(CopyToOutputDirectory)</CopyToPublishDirectory>
      </AllPublishItemsFullPathWithTargetPath>
    </ItemGroup>
  </Target>
</Project>
```

**Comment by nguerrera** [comment]
> Error MSB3094: "DestinationFiles" refers to 1 item(s), and "SourceFiles" refers to 2 item(s). They must have the same number of items.

@cceuroform That looks like a separate issue that was fixed by https://github.com/dotnet/sdk/pull/581

**Comment by nguerrera** [comment]
@cceuroform Can you do `msbuild /t:publish /flp:v=diag` and post the resulting msbuild.log to a gist. I want to verify that it is #581 and then we can talk about the right workaround while waiting for #581 to be fixed in VS.

**Comment by drewid** [comment]
@nguerrera you're totally speaking greek to me.  I just want to be able to do a basic publish to and Azure Website and not sure how specific and granular the publishing controls are to Azure.

**Comment by gulbanana** [comment]
The idea is that if you create the Directory.Build.targets file @nguerrera describes in your source code, then it will be used during the publish process, potentially restoring your ability to publish to Azure.

**Comment by cceuroform** [comment]
@nguerrera The file is huge. 116 MBytes. How can I share it with you?

**Comment by nguerrera** [comment]
@cceuroform Can you locate where the `Error MSB3094` occurs and post from that line back up to the closest occurrence of `Target "<Name>" in project "<path>"?

**Comment by cceuroform** [comment]
@nguerrera [msbuild.log](https://gist.github.com/cceuroform/c2241c54b59ddf30d80669dd5eab4e37)
I found a smaller project

**Comment by cceuroform** [comment]
@nguerrera Did you verify it is  #581?. Any workaround?

**Comment by nguerrera** [comment]
Pasting the relevant part here for others to see easily:

```
Target "_CopyResolvedFilesToPublishAlways: (TargetId:105)" in file "C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\MSBuild\Sdks\Microsoft.NET.Sdk\build\Microsoft.NET.Publish.targets"
<snip>
Task "Copy" (TaskId:109)
  Task Parameter:
      SourceFiles=
          C:\Users\cc\.nuget\packages\system.io\4.1.0\ref\net462\System.IO.dll
                  CopyLocal=false
                  CopyToPublishDirectory=Always
                  FusionName=System.IO, Version=4.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a
                  ImageRuntime=v4.0.30319
                  NuGetIsFrameworkReference=false
                  NuGetSourceType=Package
                  OriginalItemSpec=C:\Users\cc\.nuget\packages\system.io\4.1.0\ref\net462\System.IO.dll
                  Private=false
                  ReferenceSourceTarget=ResolveAssemblyReference
                  RelativePath=refs\System.IO.dll
                  ResolvedFrom={RawFileName}
                  Version=4.1.0.0
          C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.6.2\Facades\System.IO.dll
                  CopyLocal=false
                  CopyToPublishDirectory=Always
                  RelativePath=refs\System.IO.dll
                  ResolvedFrom=ImplicitlyExpandDesignTimeFacades
                  WinMDFile=false (TaskId:109)
  Task Parameter:DestinationFiles=bin\Debug\net462\win7-x86\publish\refs\System.IO.dll (TaskId:109)
  Task Parameter:OverwriteReadOnlyFiles=False (TaskId:109)

C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\MSBuild\Sdks\Microsoft.NET.Sdk\build\Microsoft.NET.Publish.targets(128,5): error MSB3094: "DestinationFiles" refers to 1 item(s), and "SourceFiles" refers to 2 item(s). They must have the same number of items. [C:\Users\cc\Documents\Visual Studio 2017\Projects\WebApplication12\src\WebApplication12\WebApplication12.csproj]
```

cc @eerhardt @dsplaisted  : This actually looks a little different -- there's a conflict with ImplicitlyExpandDesignTimeFacades and a package providing a newer System.IO facade.

Applying the two-line change from https://github.com/dotnet/sdk/pull/581/files#diff-ffd1a974928a37a20fbc471c040962ac (fix from #581) to C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\MSBuild\Sdks\Microsoft.NET.Sdk\build\Microsoft.NET.Publish.targets might get it past this point, but it might also hide the root cause and deploy the older ref, which might cause Razor some problems.

@dsplaisted I think you mentioned looking at conflicts with implicit facades somewhere. Thoughts on this?

**Comment by cceuroform** [comment]
@nguerrera I will try to apply the two-line change from #581

**Comment by cceuroform** [comment]
@nguerrera After apply the change from #581, the previous error disapeared. :)
But now  I got the following error:

C:\Data\EF\Repo\Projects\EF.Definition\EF.Definition.csproj : error MSB4057: The target "GetCopyToPublishDirectoryItems" does not exist in the project.

[msbuild.log(2)](https://gist.github.com/cceuroform/d8703a44668b36f08c8186e8f3f56be4)

Here is the csproj file [EF.Definition.csproj](https://gist.github.com/cceuroform/39b3061a57a9f0b117c1fd0fb14a6a5c)

**Comment by ankgupta067** [comment]
@nguerrera 
i tried adding the changes you suggested by creating build.targets in the root folder i seem to get no success with this issue, 

I am accessing a portable class library in .net core app(CrossPlatformWebApp).

solution can be accessed at 
https://drive.google.com/file/d/0B6UT1ay0CMA5S281VEhLSzZHUnM/view

i am getting below error when i try to publish to file system -

C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\MSBuild\15.0\Bin\csc.exe /noconfig /nowarn:1701,1702,1705,2008 /nostdlib+ /errorreport:prompt /warn:4 /define:DEBUG;TRACE;NETCOREAPP1_0 /errorendlocation /preferreduilang:en-US /reference:C:\Users\310182638\.nuget\packages\microsoft.aspnetcore.antiforgery\1.0.1\lib\netstandard1.3\Microsoft.AspNetCore.Antiforgery.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.aspnetcore.authorization\1.0.0\lib\netstandard1.3\Microsoft.AspNetCore.Authorization.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.aspnetcore.cors\1.0.0\lib\netstandard1.3\Microsoft.AspNetCore.Cors.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.aspnetcore.cryptography.internal\1.0.0\lib\netstandard1.3\Microsoft.AspNetCore.Cryptography.Internal.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.aspnetcore.dataprotection.abstractions\1.0.0\lib\netstandard1.3\Microsoft.AspNetCore.DataProtection.Abstractions.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.aspnetcore.dataprotection\1.0.0\lib\netstandard1.3\Microsoft.AspNetCore.DataProtection.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.aspnetcore.diagnostics.abstractions\1.0.0\lib\netstandard1.0\Microsoft.AspNetCore.Diagnostics.Abstractions.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.aspnetcore.hosting.abstractions\1.0.0\lib\netstandard1.3\Microsoft.AspNetCore.Hosting.Abstractions.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.aspnetcore.hosting\1.0.0\lib\netstandard1.3\Microsoft.AspNetCore.Hosting.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.aspnetcore.hosting.server.abstractions\1.0.0\lib\netstandard1.3\Microsoft.AspNetCore.Hosting.Server.Abstractions.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.aspnetcore.html.abstractions\1.0.0\lib\netstandard1.0\Microsoft.AspNetCore.Html.Abstractions.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.aspnetcore.http.abstractions\1.0.0\lib\netstandard1.3\Microsoft.AspNetCore.Http.Abstractions.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.aspnetcore.http\1.0.0\lib\netstandard1.3\Microsoft.AspNetCore.Http.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.aspnetcore.http.extensions\1.0.0\lib\netstandard1.3\Microsoft.AspNetCore.Http.Extensions.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.aspnetcore.http.features\1.0.0\lib\netstandard1.3\Microsoft.AspNetCore.Http.Features.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.aspnetcore.httpoverrides\1.0.0\lib\netstandard1.3\Microsoft.AspNetCore.HttpOverrides.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.aspnetcore.jsonpatch\1.0.0\lib\netstandard1.1\Microsoft.AspNetCore.JsonPatch.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.aspnetcore.localization\1.0.0\lib\netstandard1.3\Microsoft.AspNetCore.Localization.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.aspnetcore.mvc.abstractions\1.0.1\lib\netstandard1.3\Microsoft.AspNetCore.Mvc.Abstractions.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.aspnetcore.mvc.apiexplorer\1.0.1\lib\netstandard1.6\Microsoft.AspNetCore.Mvc.ApiExplorer.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.aspnetcore.mvc.core\1.0.1\lib\netstandard1.6\Microsoft.AspNetCore.Mvc.Core.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.aspnetcore.mvc.cors\1.0.1\lib\netstandard1.6\Microsoft.AspNetCore.Mvc.Cors.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.aspnetcore.mvc.dataannotations\1.0.1\lib\netstandard1.6\Microsoft.AspNetCore.Mvc.DataAnnotations.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.aspnetcore.mvc\1.0.1\lib\netstandard1.6\Microsoft.AspNetCore.Mvc.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.aspnetcore.mvc.formatters.json\1.0.1\lib\netstandard1.6\Microsoft.AspNetCore.Mvc.Formatters.Json.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.aspnetcore.mvc.localization\1.0.1\lib\netstandard1.6\Microsoft.AspNetCore.Mvc.Localization.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.aspnetcore.mvc.razor\1.0.1\lib\netstandard1.6\Microsoft.AspNetCore.Mvc.Razor.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.aspnetcore.mvc.razor.host\1.0.1\lib\netstandard1.6\Microsoft.AspNetCore.Mvc.Razor.Host.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.aspnetcore.mvc.taghelpers\1.0.1\lib\netstandard1.6\Microsoft.AspNetCore.Mvc.TagHelpers.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.aspnetcore.mvc.viewfeatures\1.0.1\lib\netstandard1.6\Microsoft.AspNetCore.Mvc.ViewFeatures.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.aspnetcore.razor\1.0.0\lib\netstandard1.3\Microsoft.AspNetCore.Razor.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.aspnetcore.razor.runtime\1.0.0\lib\netstandard1.5\Microsoft.AspNetCore.Razor.Runtime.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.aspnetcore.routing.abstractions\1.0.1\lib\netstandard1.3\Microsoft.AspNetCore.Routing.Abstractions.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.aspnetcore.routing\1.0.1\lib\netstandard1.3\Microsoft.AspNetCore.Routing.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.aspnetcore.server.iisintegration\1.0.0\lib\netstandard1.3\Microsoft.AspNetCore.Server.IISIntegration.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.aspnetcore.server.kestrel\1.0.1\lib\netstandard1.3\Microsoft.AspNetCore.Server.Kestrel.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.aspnetcore.webutilities\1.0.0\lib\netstandard1.3\Microsoft.AspNetCore.WebUtilities.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.codeanalysis.csharp\1.3.0\lib\netstandard1.3\Microsoft.CodeAnalysis.CSharp.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.codeanalysis.common\1.3.0\lib\netstandard1.3\Microsoft.CodeAnalysis.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.csharp\4.0.1\ref\netstandard1.0\Microsoft.CSharp.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.dotnet.internalabstractions\1.0.0\lib\netstandard1.3\Microsoft.DotNet.InternalAbstractions.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.extensions.caching.abstractions\1.0.0\lib\netstandard1.0\Microsoft.Extensions.Caching.Abstractions.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.extensions.caching.memory\1.0.0\lib\netstandard1.3\Microsoft.Extensions.Caching.Memory.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.extensions.configuration.abstractions\1.0.0\lib\netstandard1.0\Microsoft.Extensions.Configuration.Abstractions.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.extensions.configuration.binder\1.0.0\lib\netstandard1.1\Microsoft.Extensions.Configuration.Binder.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.extensions.configuration.commandline\1.0.0\lib\netstandard1.1\Microsoft.Extensions.Configuration.CommandLine.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.extensions.configuration\1.0.0\lib\netstandard1.1\Microsoft.Extensions.Configuration.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.extensions.configuration.environmentvariables\1.0.0\lib\netstandard1.3\Microsoft.Extensions.Configuration.EnvironmentVariables.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.extensions.configuration.fileextensions\1.0.0\lib\netstandard1.3\Microsoft.Extensions.Configuration.FileExtensions.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.extensions.configuration.json\1.0.0\lib\netstandard1.3\Microsoft.Extensions.Configuration.Json.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.extensions.dependencyinjection.abstractions\1.0.0\lib\netstandard1.0\Microsoft.Extensions.DependencyInjection.Abstractions.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.extensions.dependencyinjection\1.0.0\lib\netstandard1.1\Microsoft.Extensions.DependencyInjection.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.extensions.dependencymodel\1.0.0\lib\netstandard1.6\Microsoft.Extensions.DependencyModel.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.extensions.fileproviders.abstractions\1.0.0\lib\netstandard1.0\Microsoft.Extensions.FileProviders.Abstractions.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.extensions.fileproviders.composite\1.0.0\lib\netstandard1.0\Microsoft.Extensions.FileProviders.Composite.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.extensions.fileproviders.physical\1.0.0\lib\netstandard1.3\Microsoft.Extensions.FileProviders.Physical.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.extensions.filesystemglobbing\1.0.0\lib\netstandard1.3\Microsoft.Extensions.FileSystemGlobbing.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.extensions.globalization.cultureinfocache\1.0.0\lib\netstandard1.1\Microsoft.Extensions.Globalization.CultureInfoCache.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.extensions.localization.abstractions\1.0.0\lib\netstandard1.0\Microsoft.Extensions.Localization.Abstractions.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.extensions.localization\1.0.0\lib\netstandard1.3\Microsoft.Extensions.Localization.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.extensions.logging.abstractions\1.0.0\lib\netstandard1.1\Microsoft.Extensions.Logging.Abstractions.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.extensions.logging.console\1.0.0\lib\netstandard1.3\Microsoft.Extensions.Logging.Console.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.extensions.logging.debug\1.0.0\lib\netstandard1.3\Microsoft.Extensions.Logging.Debug.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.extensions.logging\1.0.0\lib\netstandard1.1\Microsoft.Extensions.Logging.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.extensions.objectpool\1.0.0\lib\netstandard1.3\Microsoft.Extensions.ObjectPool.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.extensions.options.configurationextensions\1.0.0\lib\netstandard1.1\Microsoft.Extensions.Options.ConfigurationExtensions.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.extensions.options\1.0.0\lib\netstandard1.0\Microsoft.Extensions.Options.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.extensions.platformabstractions\1.0.0\lib\netstandard1.3\Microsoft.Extensions.PlatformAbstractions.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.extensions.primitives\1.0.0\lib\netstandard1.0\Microsoft.Extensions.Primitives.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.extensions.webencoders\1.0.0\lib\netstandard1.0\Microsoft.Extensions.WebEncoders.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.net.http.headers\1.0.0\lib\netstandard1.1\Microsoft.Net.Http.Headers.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.visualbasic\10.0.1\ref\netstandard1.1\Microsoft.VisualBasic.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.win32.primitives\4.0.1\ref\netstandard1.3\Microsoft.Win32.Primitives.dll /reference:C:\Users\310182638\.nuget\packages\microsoft.win32.registry\4.0.0\ref\netstandard1.3\Microsoft.Win32.Registry.dll /reference:C:\Users\310182638\.nuget\packages\newtonsoft.json\9.0.1\lib\netstandard1.0\Newtonsoft.Json.dll /reference:"C:\Users\310182638\documents\visual studio 2017\Projects\CrossPlatformDevelopment\ClassLibrary1\bin\Release\PortableClassLibrary.dll" /reference:C:\Users\310182638\.nuget\packages\system.appcontext\4.1.0\ref\netstandard1.6\System.AppContext.dll /reference:C:\Users\310182638\.nuget\packages\system.buffers\4.0.0\lib\netstandard1.1\System.Buffers.dll /reference:C:\Users\310182638\.nuget\packages\system.collections.concurrent\4.0.12\ref\netstandard1.3\System.Collections.Concurrent.dll /reference:C:\Users\310182638\.nuget\packages\system.collections\4.0.11\ref\netstandard1.3\System.Collections.dll /reference:C:\Users\310182638\.nuget\packages\system.collections.immutable\1.2.0\lib\netstandard1.0\System.Collections.Immutable.dll /reference:C:\Users\310182638\.nuget\packages\system.collections.nongeneric\4.0.1\ref\netstandard1.3\System.Collections.NonGeneric.dll /reference:C:\Users\310182638\.nuget\packages\system.componentmodel.annotations\4.1.0\ref\netstandard1.4\System.ComponentModel.Annotations.dll /reference:C:\Users\310182638\.nuget\packages\system.componentmodel\4.0.1\ref\netstandard1.0\System.ComponentModel.dll /reference:C:\Users\310182638\.nuget\packages\system.componentmodel.primitives\4.1.0\ref\netstandard1.0\System.ComponentModel.Primitives.dll /reference:C:\Users\310182638\.nuget\packages\system.componentmodel.typeconverter\4.1.0\ref\netstandard1.5\System.ComponentModel.TypeConverter.dll /reference:C:\Users\310182638\.nuget\packages\system.console\4.0.0\ref\netstandard1.3\System.Console.dll /reference:C:\Users\310182638\.nuget\packages\system.diagnostics.contracts\4.0.1\ref\netstandard1.0\System.Diagnostics.Contracts.dll /reference:C:\Users\310182638\.nuget\packages\system.diagnostics.debug\4.0.11\ref\netstandard1.3\System.Diagnostics.Debug.dll /reference:C:\Users\310182638\.nuget\packages\system.diagnostics.diagnosticsource\4.0.0\lib\netstandard1.3\System.Diagnostics.DiagnosticSource.dll /reference:C:\Users\310182638\.nuget\packages\system.diagnostics.process\4.1.0\ref\netstandard1.4\System.Diagnostics.Process.dll /reference:C:\Users\310182638\.nuget\packages\system.diagnostics.stacktrace\4.0.1\ref\netstandard1.3\System.Diagnostics.StackTrace.dll /reference:C:\Users\310182638\.nuget\packages\system.diagnostics.tools\4.0.1\ref\netstandard1.0\System.Diagnostics.Tools.dll /reference:C:\Users\310182638\.nuget\packages\system.diagnostics.tracing\4.1.0\ref\netstandard1.5\System.Diagnostics.Tracing.dll /reference:C:\Users\310182638\.nuget\packages\system.dynamic.runtime\4.0.11\ref\netstandard1.3\System.Dynamic.Runtime.dll /reference:C:\Users\310182638\.nuget\packages\system.globalization.calendars\4.0.1\ref\netstandard1.3\System.Globalization.Calendars.dll /reference:C:\Users\310182638\.nuget\packages\system.globalization\4.0.11\ref\netstandard1.3\System.Globalization.dll /reference:C:\Users\310182638\.nuget\packages\system.globalization.extensions\4.0.1\ref\netstandard1.3\System.Globalization.Extensions.dll /reference:C:\Users\310182638\.nuget\packages\system.io.compression\4.1.0\ref\netstandard1.3\System.IO.Compression.dll /reference:C:\Users\310182638\.nuget\packages\system.io.compression.zipfile\4.0.1\ref\netstandard1.3\System.IO.Compression.ZipFile.dll /reference:C:\Users\310182638\.nuget\packages\system.io\4.1.0\ref\netstandard1.5\System.IO.dll /reference:C:\Users\310182638\.nuget\packages\system.io.filesystem\4.0.1\ref\netstandard1.3\System.IO.FileSystem.dll /reference:C:\Users\310182638\.nuget\packages\system.io.filesystem.primitives\4.0.1\ref\netstandard1.3\System.IO.FileSystem.Primitives.dll /reference:C:\Users\310182638\.nuget\packages\system.io.filesystem.watcher\4.0.0\ref\netstandard1.3\System.IO.FileSystem.Watcher.dll /reference:C:\Users\310182638\.nuget\packages\system.io.memorymappedfiles\4.0.0\ref\netstandard1.3\System.IO.MemoryMappedFiles.dll /reference:C:\Users\310182638\.nuget\packages\system.io.unmanagedmemorystream\4.0.1\ref\netstandard1.3\System.IO.UnmanagedMemoryStream.dll /reference:C:\Users\310182638\.nuget\packages\system.linq\4.1.0\ref\netstandard1.6\System.Linq.dll /reference:C:\Users\310182638\.nuget\packages\system.linq.expressions\4.1.0\ref\netstandard1.6\System.Linq.Expressions.dll /reference:C:\Users\310182638\.nuget\packages\system.linq.parallel\4.0.1\ref\netstandard1.1\System.Linq.Parallel.dll /reference:C:\Users\310182638\.nuget\packages\system.linq.queryable\4.0.1\ref\netstandard1.0\System.Linq.Queryable.dll /reference:C:\Users\310182638\.nuget\packages\system.net.http\4.1.0\ref\netstandard1.3\System.Net.Http.dll /reference:C:\Users\310182638\.nuget\packages\system.net.nameresolution\4.0.0\ref\netstandard1.3\System.Net.NameResolution.dll /reference:C:\Users\310182638\.nuget\packages\system.net.primitives\4.0.11\ref\netstandard1.3\System.Net.Primitives.dll /reference:C:\Users\310182638\.nuget\packages\system.net.requests\4.0.11\ref\netstandard1.3\System.Net.Requests.dll /reference:C:\Users\310182638\.nuget\packages\system.net.security\4.0.0\ref\netstandard1.3\System.Net.Security.dll /reference:C:\Users\310182638\.nuget\packages\system.net.sockets\4.1.0\ref\netstandard1.3\System.Net.Sockets.dll /reference:C:\Users\310182638\.nuget\packages\system.net.webheadercollection\4.0.1\ref\netstandard1.3\System.Net.WebHeaderCollection.dll /reference:C:\Users\310182638\.nuget\packages\system.net.websockets\4.0.0\ref\netstandard1.3\System.Net.WebSockets.dll /reference:C:\Users\310182638\.nuget\packages\system.numerics.vectors\4.1.1\ref\netstandard1.0\System.Numerics.Vectors.dll /reference:C:\Users\310182638\.nuget\packages\system.objectmodel\4.0.12\ref\netstandard1.3\System.ObjectModel.dll /reference:C:\Users\310182638\.nuget\packages\system.reflection.dispatchproxy\4.0.1\ref\netstandard1.3\System.Reflection.DispatchProxy.dll /reference:C:\Users\310182638\.nuget\packages\system.reflection\4.1.0\ref\netstandard1.5\System.Reflection.dll /reference:C:\Users\310182638\.nuget\packages\system.reflection.extensions\4.0.1\ref\netstandard1.0\System.Reflection.Extensions.dll /reference:C:\Users\310182638\.nuget\packages\system.reflection.metadata\1.3.0\lib\netstandard1.1\System.Reflection.Metadata.dll /reference:C:\Users\310182638\.nuget\packages\system.reflection.primitives\4.0.1\ref\netstandard1.0\System.Reflection.Primitives.dll /reference:C:\Users\310182638\.nuget\packages\system.reflection.typeextensions\4.1.0\ref\netstandard1.5\System.Reflection.TypeExtensions.dll /reference:C:\Users\310182638\.nuget\packages\system.resources.reader\4.0.0\lib\netstandard1.0\System.Resources.Reader.dll /reference:C:\Users\310182638\.nuget\packages\system.resources.resourcemanager\4.0.1\ref\netstandard1.0\System.Resources.ResourceManager.dll /reference:C:\Users\310182638\.nuget\packages\system.runtime\4.1.0\ref\netstandard1.5\System.Runtime.dll /reference:C:\Users\310182638\.nuget\packages\system.runtime.extensions\4.1.0\ref\netstandard1.5\System.Runtime.Extensions.dll /reference:C:\Users\310182638\.nuget\packages\system.runtime.handles\4.0.1\ref\netstandard1.3\System.Runtime.Handles.dll /reference:C:\Users\310182638\.nuget\packages\system.runtime.interopservices\4.1.0\ref\netstandard1.5\System.Runtime.InteropServices.dll /reference:C:\Users\310182638\.nuget\packages\system.runtime.interopservices.runtimeinformation\4.0.0\ref\netstandard1.1\System.Runtime.InteropServices.RuntimeInformation.dll /reference:C:\Users\310182638\.nuget\packages\system.runtime.loader\4.0.0\ref\netstandard1.5\System.Runtime.Loader.dll /reference:C:\Users\310182638\.nuget\packages\system.runtime.numerics\4.0.1\ref\netstandard1.1\System.Runtime.Numerics.dll /reference:C:\Users\310182638\.nuget\packages\system.runtime.serialization.primitives\4.1.1\ref\netstandard1.3\System.Runtime.Serialization.Primitives.dll /reference:C:\Users\310182638\.nuget\packages\system.security.claims\4.0.1\ref\netstandard1.3\System.Security.Claims.dll /reference:C:\Users\310182638\.nuget\packages\system.security.cryptography.algorithms\4.2.0\ref\netstandard1.6\System.Security.Cryptography.Algorithms.dll /reference:C:\Users\310182638\.nuget\packages\system.security.cryptography.encoding\4.0.0\ref\netstandard1.3\System.Security.Cryptography.Encoding.dll /reference:C:\Users\310182638\.nuget\packages\system.security.cryptography.primitives\4.0.0\ref\netstandard1.3\System.Security.Cryptography.Primitives.dll /reference:C:\Users\310182638\.nuget\packages\system.security.cryptography.x509certificates\4.1.0\ref\netstandard1.4\System.Security.Cryptography.X509Certificates.dll /reference:C:\Users\310182638\.nuget\packages\system.security.principal\4.0.1\ref\netstandard1.0\System.Security.Principal.dll /reference:C:\Users\310182638\.nuget\packages\system.security.principal.windows\4.0.0\ref\netstandard1.3\System.Security.Principal.Windows.dll /reference:C:\Users\310182638\.nuget\packages\system.text.encoding\4.0.11\ref\netstandard1.3\System.Text.Encoding.dll /reference:C:\Users\310182638\.nuget\packages\system.text.encoding.extensions\4.0.11\ref\netstandard1.3\System.Text.Encoding.Extensions.dll /reference:C:\Users\310182638\.nuget\packages\system.text.encodings.web\4.0.0\lib\netstandard1.0\System.Text.Encodings.Web.dll /reference:C:\Users\310182638\.nuget\packages\system.text.regularexpressions\4.1.0\ref\netstandard1.6\System.Text.RegularExpressions.dll /reference:C:\Users\310182638\.nuget\packages\system.threading\4.0.11\ref\netstandard1.3\System.Threading.dll /reference:C:\Users\310182638\.nuget\packages\system.threading.tasks.dataflow\4.6.0\lib\netstandard1.1\System.Threading.Tasks.Dataflow.dll /reference:C:\Users\310182638\.nuget\packages\system.threading.tasks\4.0.11\ref\netstandard1.3\System.Threading.Tasks.dll /reference:C:\Users\310182638\.nuget\packages\system.threading.tasks.extensions\4.0.0\lib\netstandard1.0\System.Threading.Tasks.Extensions.dll /reference:C:\Users\310182638\.nuget\packages\system.threading.tasks.parallel\4.0.1\ref\netstandard1.1\System.Threading.Tasks.Parallel.dll /reference:C:\Users\310182638\.nuget\packages\system.threading.thread\4.0.0\ref\netstandard1.3\System.Threading.Thread.dll /reference:C:\Users\310182638\.nuget\packages\system.threading.threadpool\4.0.10\ref\netstandard1.3\System.Threading.ThreadPool.dll /reference:C:\Users\310182638\.nuget\packages\system.threading.timer\4.0.1\ref\netstandard1.2\System.Threading.Timer.dll /reference:C:\Users\310182638\.nuget\packages\system.xml.readerwriter\4.0.11\ref\netstandard1.3\System.Xml.ReaderWriter.dll /reference:C:\Users\310182638\.nuget\packages\system.xml.xdocument\4.0.11\ref\netstandard1.3\System.Xml.XDocument.dll /debug+ /debug:portable /filealign:512 /nologo /optimize- /out:obj\Debug\netcoreapp1.0\CrossPlatformWebApp.dll /ruleset:"C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\Team Tools\Static Analysis Tools\\Rule Sets\MinimumRecommendedRules.ruleset" /target:exe /utf8output /deterministic+ /analyzer:C:\Users\310182638\.nuget\packages\microsoft.codeanalysis.analyzers\1.1.0\analyzers\dotnet\cs\Microsoft.CodeAnalysis.Analyzers.dll /analyzer:C:\Users\310182638\.nuget\packages\microsoft.codeanalysis.analyzers\1.1.0\analyzers\dotnet\cs\Microsoft.CodeAnalysis.CSharp.Analyzers.dll Controllers\ValuesController.cs Program.cs Startup.cs "C:\TEMP\.NETCoreApp,Version=v1.0.AssemblyAttributes.cs" obj\Debug\netcoreapp1.0\\TemporaryGeneratedFile_E7A71F73-0F8D-4B9B-B56E-8E70B10BC5D3.cs obj\Debug\netcoreapp1.0\\TemporaryGeneratedFile_036C0B5B-1481-4323-8D20-8F5ADCB23D92.cs obj\Debug\netcoreapp1.0\\TemporaryGeneratedFile_5937a670-0e60-4077-877b-f7221da3dda1.cs obj\Debug\netcoreapp1.0\CrossPlatformWebApp.AssemblyInfo.cs
CrossPlatformWebApp -> C:\Users\310182638\documents\visual studio 2017\Projects\CrossPlatformDevelopment\CrossPlatformWebApp\bin\Debug\netcoreapp1.0\CrossPlatformWebApp.dll
C:\Users\310182638\documents\visual studio 2017\Projects\CrossPlatformDevelopment\ClassLibrary1\PortableClasslibrary.csproj(0,0): Error MSB4057: The target "GetCopyToPublishDirectoryItems" does not exist in the project.

**Comment by cceuroform** [comment]
@ankgupta067 Try to add < Target Name="GetCopyToPublishDirectoryItems" / > in every csproj file that fails with the error. That was helping in my case.

**Comment by nguerrera** [comment]
@ankgupta067 You named the file "build.targets", but it needs to be named "Directory.build.targets" to be picked up by msbuild. Try that.

@cceuroform That would get past the error, but can also cause required files to be missing from the publish directory. The target should match https://github.com/dotnet/sdk/issues/543#issuecomment-271656546. You could put that in every project, but using Directory.build.targets makes it easier to apply to all projects and also easier to revert once the real fix lands.

**Comment by ankgupta067** [comment]
@nguerrera works perfectly thanks

**Comment by srivatsn** [comment]
The fix for this has been checked into MSBuild - https://github.com/Microsoft/msbuild/pull/1544

**Comment by ConX-Ryan** [comment]
Thanks for working on the problem ... seems there is a clear solution now, how does the public get there hands on an update?
